api/well_inventory.py

```diff
@@ -0,0 +1,203 @@
+# ===============================================================================
+# Copyright 2025 ross
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+import csv
+from datetime import datetime
+from io import StringIO
+from itertools import groupby
+from typing import Optional, Set
+
+from fastapi import APIRouter, UploadFile, File
+from fastapi.responses import JSONResponse
+from pydantic import BaseModel, ValidationError
+from shapely import Point
+from sqlalchemy import select
+
+from constants import SRID_UTM_ZONE_13N, SRID_UTM_ZONE_12N, SRID_WGS84
+from core.dependencies import session_dependency
+from core.enums import ContactType, Role, ElevationMethod
+from db import (
+    Group,
+    ThingIdLink,
+    GroupThingAssociation,
+    Location,
+    LocationThingAssociation,
+)
+from db.thing import Thing
+from services.util import transform_srid
+
+router = APIRouter(prefix="/well-inventory-csv")
+
+
+class WellInventoryRow(BaseModel):
+    project: str
+    well_name_point_id: str
+    site_name: str
+    date_time: datetime
+    field_staff: str
+    utm_easting: float
+    utm_northing: float
+    utm_zone: int
+    elevation_ft: float
+    elevation_method: ElevationMethod
+    measuring_point_height_ft: float
+
+    # Optional lexicon fields
+    contact_role: Optional[Role] = None
+    contact_type: Optional[ContactType] = None
```

Comment:
This should go into the schemas directory (schemas/well_inventory.py) to keep the code base/style consistent